### PR TITLE
feat: added random_states func + tests

### PR DESCRIPTION
**Pull request by ankit-pn**
## Description
added random_states function and its test as described in issue #380 

## Todos
Notable points that this PR has either accomplished or will accomplish.
  -  [ ] Implemented random_state function in rand/random_states.py
  -  [ ] Implemented its test in rand/test/test_random_states.py

## Status
-  [ ] Ready to go

### CODE REVIEW

**Comment by ankit-pn** [comment]
I there any doc specifying what kind of code style that we need to follow, or which formatter I need to use to format the python code.

**Comment by purva-thakre** [comment]
@ankit-pn if you look at the output of the failing tests, it tells you what error is being caught by which formatter/linter. 

Once those are fixed, you will also have to add `qiskit` as a dependency in the following file otherwise `check-build` workflow will fail because `qiskit` can't be imported. 

https://github.com/vprusso/toqito/blob/73641d84f25ab04e31fdf657ee97c64726d17b33/pyproject.toml#L18

**Comment by ankit-pn** [comment]
> @ankit-pn if you look at the output of the failing tests, it tells you what error is being caught by which formatter/linter.
> 
> Once those are fixed, you will also have to add `qiskit` as a dependency in the following file otherwise `check-build` workflow will fail because `qiskit` can't be imported by `pytest`.
> 
> https://github.com/vprusso/toqito/blob/73641d84f25ab04e31fdf657ee97c64726d17b33/pyproject.toml#L18

Thanks Purva, I have added qiskit as dependencies in pyproject.toml and formatted the code accordingly.

**Comment by vprusso** [comment]
Also, don't forget to address the issues that the linter is catching [here](https://github.com/vprusso/toqito/actions/runs/7271316364/job/19821922692?pr=383)

Additionally, you'll need to add `toqito.rand.random_state` to `docs/states.rst` under the "States" heading. 

Feel free to ping if you have any questions!

**Comment by purva-thakre** [comment]
@ankit-pn To check if all the workflows pass locally (after all the dependencies are installed in your virtual environment)

- For ruff, use `ruff check .` in the main `toqito` directory.
- For pylint, use `poetry run pylint --fail-under=10.0 toqito/` in the main `toqito` directory
- For `pytest` use `pytest` in `toqito/toqito`

If you don't understand what an error is trying to tell you, you can look at how the function docstrings/tests are formatted in other modules.

File: toqito/rand/random_states.py
Comment (purva-thakre) [inline toqito/rand/random_states.py:31]: You do want to make sure you stay consistent with how these are listed in the docstrings of existing functions. 

For example,

https://github.com/vprusso/toqito/blob/5c1571ae483320ad40ce54d0a08f6a9bf31f2f4f/toqito/channel_ops/apply_channel.py#L93-L98

Also, make sure you add this addition to the `docs` as Vincent said in his comment.

Comment (ankit-pn) [reply toqito/rand/random_states.py:31]: specifying parameter with :param tag is failing ``` ruff check .```  . is there any way to bypass it. I am getting this error if i remove argument description that i have included in previous pr and use :param tag.

`ruff check rand/ --fix
rand/random_states.py:7:5: D417 Missing argument descriptions in the docstring for `random_states`: `d`, `n`
Found 1 error.`

Comment (purva-thakre) [reply toqito/rand/random_states.py:31]: Is your docstring following everything as expected? For example, Ruff is set to using `===` for section headings instead of `---`.

Pay attention to the full docstring I linked above and don't use an auto-formatter. FYI, we don't use `black`.

Comment (ankit-pn) [reply toqito/rand/random_states.py]: i have pushed a change, pls review it and let me if it is ok or i need to make some more change.

**Comment by vprusso** [comment]
> Also, don't forget to address the issues that the linter is catching [here](https://github.com/vprusso/toqito/actions/runs/7271316364/job/19821922692?pr=383)
> 
> Additionally, you'll need to add `toqito.rand.random_state` to `docs/states.rst` under the "States" heading.
> 
> Feel free to ping if you have any questions!

Seems like this still isn't part of the diff, @ankit-pn

**Comment by ankit-pn** [comment]
has made fixes for it, can you check? have added  ```toqito.rand.random_state``` to ```docs/rand.rst```.

**Comment by vprusso** [comment]
> has made fixes for it, can you check? have added `toqito.rand.random_state` to `docs/rand.rst`.

Yep! That's what I was suggesting. Nice!

**Review by vprusso — APPROVED** [review]
LGTM! I'll approve but want to wait for @purva-thakre to give the final sign-off. 

Great work, @ankit-pn, thank you again for your contribution!

**Comment by purva-thakre** [comment]
LGTM too! :fireworks:

**Pull request merged**
